### PR TITLE
Debug the refactoring done in feat/v1.3

### PR DESCRIPTION
**Pull request by ayulockin**
## TLDR;

This PR replicates the #64 branch with evaluation driven improvements. Evaluation is critical for LLM based systems.

## Overview:

* The PR first reproduced the main branch's evaluation results of 72%.
* Each commit from the #64 branch was carefully added followed by rigorous auto evaluation. No modifications were made to the evaluation strategy or pipeline except making it work asynchronously to speed up evaluation from few hours to under 10 mins.
* At each junction, experiments were conducted to keep the evaluation result as close as possible to the baseline.
* One of the main junction came when FAISS vector store was switched with ChromaDB and the ingestion pipeline used `text-embedding-3-small` model. Experiments were conducted to ensure these changes will not hurt the overall system a lot.
* Another important junction was switching from LlamaIndex to Langchain Expression Language. Multiple commits were incorporated to make the system work correctly. Evaluation allowed to remain close to the baseline.
* System prompt was changed in one of the commit which was one the big reason for the performance degradation. Switched it back to the old system prompt with minor modifications. Improved the performance massively.
* As we came closer to completing the PR, we were able to beat the baseline and gain almost 8% gain over the baseline.

PS: When the #64 branch was evaluated separately, the accuracy was under 30%. With evaluation driven development we were able to beat the baseline.

PSS: The documentation of every experiment is coming soon.

### CODE REVIEW

File: src/wandbot/chat/config.py [hunk 1]
```diff
@@ -21,18 +21,18 @@
 class ChatConfig(BaseSettings):
     chat_model_name: str = "gpt-4-1106-preview"
     max_retries: int = 2
-    fallback_model_name: str = "gpt-3.5-turbo-1106"
+    fallback_model_name: str = "gpt-4-1106-preview"
```
Comment (parambharat) [inline src/wandbot/chat/config.py]: @ayulockin : This fallback was supposed to be a model that can be used when GPT-4 calls fail. I know for evals this was set to 4, but we should use a fallback that is not the same model in prod.

File: src/wandbot/ingestion/__main__.py [hunk 2]
```diff
@@ -11,10 +11,17 @@ def main():
...
+    raw_artifact = "wandbot/wandbot-dev/raw_dataset:v39"
+
+    # preprocessed_artifact = preprocess_data.load(project, entity, raw_artifact)
+    preprocessed_artifact = "wandbot/wandbot-dev/transformed_data:v5"
+
+    vectorstore_artifact = vectorstores.load(
+        project, entity, preprocessed_artifact
+    )
+
+    # create_ingestion_report(project, entity, raw_artifact, vectorstore_artifact)
```
Comment (parambharat) [inline src/wandbot/ingestion/__main__.py]: @ayulockin : We should re-enable the ingestion report.

**Review by parambharat — APPROVED** [review]
@ayulockin Amazing work. Thanks you for this. Just one or two small observations.
